Godeps/_workspace/src/k8s.io/kubernetes/pkg/runtime/unstructured.go

```diff
@@ -70,10 +76,58 @@ func (unstructuredJSONScheme) DecodeInto(data []byte, obj Object) error {
 	if len(unstruct.Kind) == 0 {
 		return conversion.NewMissingKindErr(string(data))
 	}
+
+	if err := convertMapNumbers(m); err != nil {
+		return err
+	}
+
 	unstruct.Object = m
 	return nil
 }
 
+func convertMapNumbers(m map[string]interface{}) error {
+	var err error
+	for k, v := range m {
+		switch v := v.(type) {
+		case json.Number:
+			m[k], err = convertNumber(v)
+		case map[string]interface{}:
+			err = convertMapNumbers(v)
+		case []interface{}:
+			err = convertSliceNumbers(v)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func convertSliceNumbers(s []interface{}) error {
+	var err error
+	for i, v := range s {
+		switch v := v.(type) {
+		case json.Number:
+			s[i], err = convertNumber(v)
+		case map[string]interface{}:
+			err = convertMapNumbers(v)
+		case []interface{}:
+			err = convertSliceNumbers(v)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func convertNumber(n json.Number) (interface{}, error) {
+	// Attempt to convert to an int64 first
+	if i, err := n.Int64(); err == nil {
+		return i, nil
+	}
+	// Return a float64 (default json.Decode() behavior)
+	return n.Float64()
+}
```

Comment:
I think this method is functionally complete, since the only other distinct numeric type in go is `complex`, and there is no serialization scheme for complex types in json.